FreeBSD: version 115: error: unknown type name 'off_t'

FreeBSD 12.4 / Clang 13.0.0

```
===>  Configuring for binaryen-115
===>  Performing out-of-source build
/bin/mkdir -p /wrkdirs/usr/ports/devel/binaryen/work/.build
-- The C compiler identification is Clang 13.0.0
-- The CXX compiler identification is Clang 13.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building with -DBUILD_LLVM_DWARF
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Found Threads: TRUE  
-- Building with -Wall
-- Building with -Werror
-- Building with -Wextra
-- Building with -Wno-unused-parameter
-- Building with -fno-omit-frame-pointer
-- Building with -fno-rtti
-- Building with -Wno-implicit-int-float-conversion
-- Building with -Wno-unknown-warning-option
-- Building with -Wswitch
-- Building with -Wimplicit-fallthrough
-- Building with -Wnon-virtual-dtor
-- Building with -fPIC
-- Building with -UNDEBUG
-- Building with -fcolor-diagnostics
-- Building libbinaryen as shared library.
-- Configuring done (0.6s)
-- Generating done (0.0s
```
```
===>  Building for binaryen-115
[  0% 12/262] /usr/bin/c++  -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/third_party/llvm-project/include -I/wrkdirs/usr/ports/devel/binaryen/work/.build -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -DBUILD_LLVM_DWARF -Wall -Werror -Wextra -Wno-unused-parameter -fno-omit-frame-pointer -fno-rtti -Wno-implicit-int-float-conversion -Wno-unknown-warning-option -Wswitch -Wimplicit-fallthrough -Wnon-virtual-dtor -fPIC -fcolor-diagnostics -O2 -pipe -fstack-protector-strong -fno-strict-aliasing   -DNDEBUG -UNDEBUG -std=c++17 -MD -MT src/ir/CMakeFiles/ir.dir/export-utils.cpp.o -MF src/ir/CMakeFiles/ir.dir/export-utils.cpp.o.d -o src/ir/CMakeFiles/ir.dir/export-utils.cpp.o -c /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/export-utils.cpp
[  0% 13/262] /usr/bin/c++  -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/third_party/llvm-project/include -I/wrkdirs/usr/ports/devel/binaryen/work/.build -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -DBUILD_LLVM_DWARF -Wall -Werror -Wextra -Wno-unused-parameter -fno-omit-frame-pointer -fno-rtti -Wno-implicit-int-float-conversion -Wno-unknown-warning-option -Wswitch -Wimplicit-fallthrough -Wnon-virtual-dtor -fPIC -fcolor-diagnostics -O2 -pipe -fstack-protector-strong -fno-strict-aliasing   -DNDEBUG -UNDEBUG -std=c++17 -MD -MT src/ir/CMakeFiles/ir.dir/intrinsics.cpp.o -MF src/ir/CMakeFiles/ir.dir/intrinsics.cpp.o.d -o src/ir/CMakeFiles/ir.dir/intrinsics.cpp.o -c /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/intrinsics.cpp
[  1% 14/262] /usr/bin/c++  -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/third_party/llvm-project/include -I/wrkdirs/usr/ports/devel/binaryen/work/.build -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -DBUILD_LLVM_DWARF -Wall -Werror -Wextra -Wno-unused-parameter -fno-omit-frame-pointer -fno-rtti -Wno-implicit-int-float-conversion -Wno-unknown-warning-option -Wswitch -Wimplicit-fallthrough -Wnon-virtual-dtor -fPIC -fcolor-diagnostics -O2 -pipe -fstack-protector-strong -fno-strict-aliasing   -DNDEBUG -UNDEBUG -std=c++17 -MD -MT src/ir/CMakeFiles/ir.dir/names.cpp.o -MF src/ir/CMakeFiles/ir.dir/names.cpp.o.d -o src/ir/CMakeFiles/ir.dir/names.cpp.o -c /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/names.cpp
[  1% 15/262] /usr/bin/c++  -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/third_party/llvm-project/include -I/wrkdirs/usr/ports/devel/binaryen/work/.build -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -DBUILD_LLVM_DWARF -Wall -Werror -Wextra -Wno-unused-parameter -fno-omit-frame-pointer -fno-rtti -Wno-implicit-int-float-conversion -Wno-unknown-warning-option -Wswitch -Wimplicit-fallthrough -Wnon-virtual-dtor -fPIC -fcolor-diagnostics -O2 -pipe -fstack-protector-strong -fno-strict-aliasing   -DNDEBUG -UNDEBUG -std=c++17 -MD -MT src/ir/CMakeFiles/ir.dir/eh-utils.cpp.o -MF src/ir/CMakeFiles/ir.dir/eh-utils.cpp.o.d -o src/ir/CMakeFiles/ir.dir/eh-utils.cpp.o -c /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.cpp
FAILED: src/ir/CMakeFiles/ir.dir/eh-utils.cpp.o 
/usr/bin/c++  -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src -I/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/third_party/llvm-project/include -I/wrkdirs/usr/ports/devel/binaryen/work/.build -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -DBUILD_LLVM_DWARF -Wall -Werror -Wextra -Wno-unused-parameter -fno-omit-frame-pointer -fno-rtti -Wno-implicit-int-float-conversion -Wno-unknown-warning-option -Wswitch -Wimplicit-fallthrough -Wnon-virtual-dtor -fPIC -fcolor-diagnostics -O2 -pipe -fstack-protector-strong -fno-strict-aliasing   -DNDEBUG -UNDEBUG -std=c++17 -MD -MT src/ir/CMakeFiles/ir.dir/eh-utils.cpp.o -MF src/ir/CMakeFiles/ir.dir/eh-utils.cpp.o.d -o src/ir/CMakeFiles/ir.dir/eh-utils.cpp.o -c /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.cpp
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.cpp:17:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.h:20:
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:188:5: error: unknown type name 'off_t'
    off_t operator-(const Iterator& other) const { return index - other.index; }
    ^
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.cpp:17:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.h:21:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/wasm.h:36:
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/literal.h:778:21: error: invalid operands to binary expression ('wasm::SmallVector<wasm::Literal, 1>::Iterator' and 'wasm::SmallVector<wasm::Literal, 1>::Iterator')
        for (auto c : values) {
                    ^
/usr/include/c++/v1/__variant/monostate.h:46:16: note: candidate function not viable: no known conversion from 'wasm::SmallVector<wasm::Literal, 1>::Iterator' to 'std::monostate' for 1st argument
constexpr bool operator!=(monostate, monostate) noexcept { return false; }
               ^
/usr/include/c++/v1/system_error:420:1: note: candidate function not viable: no known conversion from 'wasm::SmallVector<wasm::Literal, 1>::Iterator' to 'const std::error_code' for 1st argument
operator!=(const error_code& __x, const error_code& __y) _NOEXCEPT
^
/usr/include/c++/v1/system_error:425:1: note: candidate function not viable: no known conversion from 'wasm::SmallVector<wasm::Literal, 1>::Iterator' to 'const std::error_code' for 1st argument
operator!=(const error_code& __x, const error_condition& __y) _NOEXCEPT
^
/usr/include/c++/v1/system_error:430:1: note: candidate function not viable: no known conversion from 'wasm::SmallVector<wasm::Literal, 1>::Iterator' to 'const std::error_condition' for 1st argument
operator!=(const error_condition& __x, const error_code& __y) _NOEXCEPT
^
/usr/include/c++/v1/system_error:435:1: note: candidate function not viable: no known conversion from 'wasm::SmallVector<wasm::Literal, 1>::Iterator' to 'const std::error_condition' for 1st argument
operator!=(const error_condition& __x, const error_condition& __y) _NOEXCEPT
^
/usr/include/c++/v1/__utility/pair.h:329:1: note: candidate template ignored: could not match 'pair<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const pair<_T1,_T2>& __x, const pair<_T1,_T2>& __y)
^
/usr/include/c++/v1/tuple:1306:1: note: candidate template ignored: could not match 'tuple<type-parameter-0-0...>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const tuple<_Tp...>& __x, const tuple<_Up...>& __y)
^
/usr/include/c++/v1/__memory/allocator.h:248:6: note: candidate template ignored: could not match 'allocator<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(const allocator<_Tp>&, const allocator<_Up>&) _NOEXCEPT {return false;}
     ^
/usr/include/c++/v1/__memory/unique_ptr.h:577:1: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const unique_ptr<_T1, _D1>& __x, const unique_ptr<_T2, _D2>& __y) {return !(__x == __y);}
^
/usr/include/c++/v1/__memory/unique_ptr.h:624:1: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const unique_ptr<_T1, _D1>& __x, nullptr_t) _NOEXCEPT
^
/usr/include/c++/v1/__memory/unique_ptr.h:632:1: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(nullptr_t, const unique_ptr<_T1, _D1>& __x) _NOEXCEPT
^
/usr/include/c++/v1/__memory/shared_ptr.h:1129:1: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const shared_ptr<_Tp>& __x, const shared_ptr<_Up>& __y) _NOEXCEPT
^
/usr/include/c++/v1/__memory/shared_ptr.h:1191:1: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const shared_ptr<_Tp>& __x, nullptr_t) _NOEXCEPT
^
/usr/include/c++/v1/__memory/shared_ptr.h:1199:1: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(nullptr_t, const shared_ptr<_Tp>& __x) _NOEXCEPT
^
/usr/include/c++/v1/variant:1623:16: note: candidate template ignored: could not match 'variant<type-parameter-0-0...>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
constexpr bool operator!=(const variant<_Types...>& __lhs,
               ^
/usr/include/c++/v1/__iterator/istreambuf_iterator.h:102:6: note: candidate template ignored: could not match 'istreambuf_iterator<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(const istreambuf_iterator<_CharT,_Traits>& __a,
     ^
/usr/include/c++/v1/__iterator/istream_iterator.h:93:1: note: candidate template ignored: could not match 'istream_iterator<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const istream_iterator<_Tp, _CharT, _Traits, _Distance>& __x,
^
/usr/include/c++/v1/__iterator/move_iterator.h:121:1: note: candidate template ignored: could not match 'move_iterator<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const move_iterator<_Iter1>& __x, const move_iterator<_Iter2>& __y)
^
/usr/include/c++/v1/__iterator/reverse_iterator.h:170:1: note: candidate template ignored: could not match 'reverse_iterator<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
^
/usr/include/c++/v1/__iterator/wrap_iter.h:205:6: note: candidate template ignored: could not match '__wrap_iter<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter1>& __y) _NOEXCEPT
     ^
/usr/include/c++/v1/__iterator/wrap_iter.h:212:6: note: candidate template ignored: could not match '__wrap_iter<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter2>& __y) _NOEXCEPT
     ^
/usr/include/c++/v1/__functional/function.h:1218:1: note: candidate template ignored: could not match 'function<type-parameter-0-0 (type-parameter-0-1...)>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const function<_Rp(_ArgTypes...)>& __f, nullptr_t) _NOEXCEPT {return (bool)__f;}
^
/usr/include/c++/v1/__functional/function.h:1223:1: note: candidate template ignored: could not match 'function<type-parameter-0-0 (type-parameter-0-1...)>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(nullptr_t, const function<_Rp(_ArgTypes...)>& __f) _NOEXCEPT {return (bool)__f;}
^
/usr/include/c++/v1/array:378:1: note: candidate template ignored: could not match 'array<type-parameter-0-0, _Size>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const array<_Tp, _Size>& __x, const array<_Tp, _Size>& __y)
^
/usr/include/c++/v1/vector:3368:1: note: candidate template ignored: could not match 'vector<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const vector<_Tp, _Allocator>& __x, const vector<_Tp, _Allocator>& __y)
^
/usr/include/c++/v1/optional:1057:1: note: candidate template ignored: could not match 'optional<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const optional<_Tp>& __x, const optional<_Up>& __y)
^
/usr/include/c++/v1/optional:1150:1: note: candidate template ignored: could not match 'optional<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const optional<_Tp>& __x, nullopt_t) noexcept
^
/usr/include/c++/v1/optional:1158:1: note: candidate template ignored: could not match 'optional<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(nullopt_t, const optional<_Tp>& __x) noexcept
^
/usr/include/c++/v1/optional:1259:1: note: candidate template ignored: could not match 'optional<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const optional<_Tp>& __x, const _Up& __v)
^
/usr/include/c++/v1/optional:1271:1: note: candidate template ignored: could not match 'optional<type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const _Tp& __v, const optional<_Up>& __x)
^
/usr/include/c++/v1/map:1643:1: note: candidate template ignored: could not match 'map<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const map<_Key, _Tp, _Compare, _Allocator>& __x,
^
/usr/include/c++/v1/map:2241:1: note: candidate template ignored: could not match 'multimap<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const multimap<_Key, _Tp, _Compare, _Allocator>& __x,
^
/usr/include/c++/v1/string_view:706:6: note: candidate template ignored: could not match 'basic_string_view<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(basic_string_view<_CharT, _Traits> __lhs, basic_string_view<_CharT, _Traits> __rhs) _NOEXCEPT
     ^
/usr/include/c++/v1/string_view:715:6: note: candidate template ignored: could not match 'basic_string_view<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(basic_string_view<_CharT, _Traits> __lhs,
     ^
/usr/include/c++/v1/string_view:725:6: note: candidate template ignored: could not match 'basic_string_view<type-parameter-0-0, type-parameter-0-1>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(typename common_type<basic_string_view<_CharT, _Traits> >::type __lhs,
     ^
/usr/include/c++/v1/string:586:6: note: candidate template ignored: could not match 'fpos<type-parameter-0-0>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
bool operator!=(const fpos<_StateT>& __x, const fpos<_StateT>& __y)
     ^
/usr/include/c++/v1/string:4116:1: note: candidate template ignored: could not match 'basic_string<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const basic_string<_CharT,_Traits,_Allocator>& __lhs,
^
/usr/include/c++/v1/string:4125:1: note: candidate template ignored: could not match 'const _CharT *' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const _CharT* __lhs,
^
/usr/include/c++/v1/string:4134:1: note: candidate template ignored: could not match 'basic_string<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const basic_string<_CharT, _Traits, _Allocator>& __lhs,
^
/usr/include/c++/v1/unordered_map:1849:1: note: candidate template ignored: could not match 'unordered_map<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3, type-parameter-0-4>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>& __x,
^
/usr/include/c++/v1/unordered_map:2586:1: note: candidate template ignored: could not match 'unordered_multimap<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3, type-parameter-0-4>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const unordered_multimap<_Key, _Tp, _Hash, _Pred, _Alloc>& __x,
^
/usr/include/c++/v1/set:936:1: note: candidate template ignored: could not match 'set<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const set<_Key, _Compare, _Allocator>& __x,
^
/usr/include/c++/v1/set:1467:1: note: candidate template ignored: could not match 'multiset<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const multiset<_Key, _Compare, _Allocator>& __x,
^
/usr/include/c++/v1/unordered_set:1101:1: note: candidate template ignored: could not match 'unordered_set<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const unordered_set<_Value, _Hash, _Pred, _Alloc>& __x,
^
/usr/include/c++/v1/unordered_set:1771:1: note: candidate template ignored: could not match 'unordered_multiset<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3>' against 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
operator!=(const unordered_multiset<_Value, _Hash, _Pred, _Alloc>& __x,
^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/literal.h:778:23: note: in implicit call to 'operator!=' for iterator of type 'wasm::Literals'
        for (auto c : values) {
                      ^~~~~~
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:207:12: note: selected 'begin' function with iterator type 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
  Iterator begin() { return Iterator(this, 0); }
           ^
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.cpp:18:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/branch-utils.h:20:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/iteration.h:20:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/properties.h:20:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/bits.h:21:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/literal-utils.h:20:
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/wasm-builder.h:1217:19: error: invalid operands to binary expression ('wasm::SmallVector<wasm::Literal, 1>::Iterator' and 'wasm::SmallVector<wasm::Literal, 1>::Iterator')
      for (auto c : value.getGCData()->values) {
                  ^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/wasm-builder.h:1217:21: note: in implicit call to 'operator!=' for iterator of type 'wasm::Literals'
      for (auto c : value.getGCData()->values) {
                    ^~~~~
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:207:12: note: selected 'begin' function with iterator type 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
  Iterator begin() { return Iterator(this, 0); }
           ^
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.cpp:18:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/branch-utils.h:20:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/iteration.h:20:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/properties.h:20:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/bits.h:21:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/literal-utils.h:20:
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/wasm-builder.h:1236:23: error: invalid operands to binary expression ('wasm::SmallVector<wasm::Literal, 1>::Iterator' and 'wasm::SmallVector<wasm::Literal, 1>::Iterator')
      for (auto value : values) {
                      ^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/wasm-builder.h:1236:25: note: in implicit call to 'operator!=' for iterator of type 'wasm::Literals'
      for (auto value : values) {
                        ^~~~~~
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:207:12: note: selected 'begin' function with iterator type 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
  Iterator begin() { return Iterator(this, 0); }
           ^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:197:9: error: type 'IteratorBase<SmallVector<wasm::Literal, 1UL>, wasm::SmallVector<wasm::Literal, 1>::Iterator>' is not a direct or virtual base of 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
      : IteratorBase<SmallVector<T, N>, Iterator>(parent, index) {}
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:207:29: note: in instantiation of member function 'wasm::SmallVector<wasm::Literal, 1>::Iterator::Iterator' requested here
  Iterator begin() { return Iterator(this, 0); }
                            ^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/literal.h:709:20: note: in instantiation of member function 'wasm::SmallVector<wasm::Literal, 1>::begin' requested here
    for (auto& val : *this) {
                   ^
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.cpp:17:
In file included from /wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/ir/eh-utils.h:20:
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:208:27: error: no matching constructor for initialization of 'wasm::SmallVector<wasm::Literal, 1>::Iterator'
  Iterator end() { return Iterator(this, size()); }
                          ^        ~~~~~~~~~~~~
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/literal.h:778:21: note: in instantiation of member function 'wasm::SmallVector<wasm::Literal, 1>::end' requested here
        for (auto c : values) {
                    ^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:195:10: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided
  struct Iterator : IteratorBase<SmallVector<T, N>, Iterator> {
         ^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:195:10: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 2 were provided
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:203:9: error: type 'IteratorBase<const SmallVector<wasm::Literal, 1UL>, wasm::SmallVector<wasm::Literal, 1>::ConstIterator>' is not a direct or virtual base of 'wasm::SmallVector<wasm::Literal, 1>::ConstIterator'
      : IteratorBase<const SmallVector<T, N>, ConstIterator>(parent, index) {}
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/support/small_vector.h:209:40: note: in instantiation of member function 'wasm::SmallVector<wasm::Literal, 1>::ConstIterator::ConstIterator' requested here
  ConstIterator begin() const { return ConstIterator(this, 0); }
                                       ^
/wrkdirs/usr/ports/devel/binaryen/work/binaryen-version_115/src/literal.h:793:26: note: in instantiation of member function 'wasm::SmallVector<wasm::Literal, 1>::begin' requested here
    for (const auto& lit : a) {
                         ^
7 errors generated.
```
Any clues?

--- a/src/support/small_vector.h
+++ b/src/support/small_vector.h
@@ -185,7 +185,7 @@ public:
       return Iterator(*this) += off;
     }
 
-    off_t operator-(const Iterator& other) const { return index - other.index; }
+    difference_type operator-(const Iterator& other) const { return index - other.index; }
 
     bool operator==(const Iterator& other) const {
       return parent == other.parent && index == other.index;

--- a/src/support/small_vector.h
+++ b/src/support/small_vector.h
@@ -185,7 +185,9 @@ public:
       return Iterator(*this) += off;
     }
 
-    difference_type operator-(const Iterator& other) const { return index - other.index; }
+    difference_type operator-(const Iterator& other) const {
+      return index - other.index;
+    }
 
     bool operator==(const Iterator& other) const {
       return parent == other.parent && index == other.index;

